features/cellular/framework/targets/UBLOX/N2XX/UBLOX_N2XX_CellularStack.cpp

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright (c) 2018, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "UBLOX_N2XX_CellularStack.h"
+#include "CellularUtil.h"
+#include <stdlib.h>
+
+using namespace mbed;
+using namespace mbed_cellular_util;
+
+UBLOX_N2XX_CellularStack::UBLOX_N2XX_CellularStack(ATHandler &atHandler, int cid, nsapi_ip_stack_t stack_type) : AT_CellularStack(atHandler, cid, stack_type)
+{
+    // URC handlers for sockets
+    _at.set_urc_handler("+NSONMI:", callback(this, &UBLOX_N2XX_CellularStack::NSONMI_URC));
+}
+
+UBLOX_N2XX_CellularStack::~UBLOX_N2XX_CellularStack()
+{
+}
+
+nsapi_error_t UBLOX_N2XX_CellularStack::socket_listen(nsapi_socket_t handle, int backlog)
+{
+    return NSAPI_ERROR_UNSUPPORTED;
+}
+
+nsapi_error_t UBLOX_N2XX_CellularStack::socket_accept(void *server, void **socket, SocketAddress *addr)
+{
+    return NSAPI_ERROR_UNSUPPORTED;
+}
+
+// Callback for Socket Read URC.
+void UBLOX_N2XX_CellularStack::NSONMI_URC()
+{
+    int a, b;
+    CellularSocket *socket;
+
+    a = _at.read_int();
+    b = _at.read_int();
+
+    socket = find_socket(a);
+    if (socket != NULL) {
+        socket->pending_bytes = b;
+        // No debug prints here as they can affect timing
+        // and cause data loss in UARTSerial
+        if (socket->_cb != NULL) {
+            socket->_cb(socket->_data);
+        }
+    }
+}
+
+int UBLOX_N2XX_CellularStack::get_max_socket_count()
+{
+    return UBLOX_MAX_SOCKET;
+}
+
+bool UBLOX_N2XX_CellularStack::is_protocol_supported(nsapi_protocol_t protocol)
+{
+    return (protocol == NSAPI_UDP);
+}
+
+nsapi_error_t UBLOX_N2XX_CellularStack::create_socket_impl(CellularSocket *socket)
+{
+    int sock_id = 0;
+    int localport = socket->localAddress.get_port();
+
+    if(localport == 5683 && localport < 0 && localport > 65535) {
```

Comment:
If local port is == 5683 but it is neither  less than 0 nor greater than 65535. Will this condition be ever true ?

Reply:
Why only prohibition of 5683 port ?

Reply:
According to AT command manual 
<listen_port>: Local port that will be included in sent messages and on which messages will be
received. The range goes from 0 to 65535 except for 5683.

Reply:
Will this condition be ever true ?

Reply:
Updated.